core/src/main/scala/cromwell/core/logging/LoggerWrapper.scala

```diff
@@ -0,0 +1,220 @@
+package cromwell.core.logging
+
+import akka.event.LoggingAdapter
+import org.apache.commons.lang3.exception.ExceptionUtils
+import org.slf4j.Logger
+import org.slf4j.helpers.MarkerIgnoringBase
+
+/**
+  * Wraps an akka logger and a Set of slf4j loggers together.
+  */
+abstract class LoggerWrapper extends MarkerIgnoringBase {
+
+  def akkaLogger: Option[LoggingAdapter]
+  def slf4jLoggers: Set[Logger]
+
+  def tag: String
+
+  private def format(msg: String): String = s"$tag: $msg"
+
+  private def format(msg: String, throwable: Throwable): String = {
+    format(msg) + ":\n" + ExceptionUtils.getStackTrace(throwable)
+  }
+
+  /**
+    * Akka logger only supports 1 to 4 arguments, unlike sl4j logger which support arbitrary number via varargs.
+    * This method matches the varargs to the correct akka logger method, if it exists.
+    */
+  private def varargsAkkaLog(pattern: String, arguments: Any*): Unit = {
+    lazy val formatted: String = format(pattern)
+
+    arguments.length match {
+      case 1 => akkaLogger.foreach(_.error(formatted, arguments.head))
+      case 2 => akkaLogger.foreach(_.error(formatted, arguments.head, arguments(1)))
+      case 3 => akkaLogger.foreach(_.error(formatted, arguments.head, arguments(1), arguments(2)))
+      case 4 => akkaLogger.foreach(_.error(formatted, arguments.head, arguments(1), arguments(2), arguments(3)))
+      case _ =>
```

Comment:
Possibly yes, I mostly wanted to avoid cases where one would use one of those varargs methods and not see anything in the log and wonder wth is happening.